resolver/resolver.cc

```diff
@@ -295,10 +301,52 @@ class ResolveConstantsWalk {
         return !checker.seenUnresolved;
     }
 
-    static core::SymbolRef resolveConstant(core::Context ctx, const shared_ptr<Nesting> &nesting,
-                                           const ast::UnresolvedConstantLit &c, bool &resolutionFailed) {
+    // Find load-time out-of-order references.
+    // Algorithm:
+    // if resolved symbol is only defined in the current file, and
+    //    the definition is after the reference loc,
+    // then, report an error.
+    static void
+    checkReferenceOrder(core::Context ctx, core::SymbolRef resolutionResult, const ast::UnresolvedConstantLit &c,
+                        const shared_ptr<UnorderedMap<core::SymbolRef, core::LocOffsets>> firstDefinitionLocs) {
```

Comment:
I vote for either taking this parameter by reference, or changing it to `const UnorderedMap<core::SymbolRef, core::LocOffsets> &` and changing the callers.  Your choice.